cep-??.md

```diff
@@ -0,0 +1,108 @@
+<table>
+<tr><td> Title </td><td> Environment yaml specification </td>
+<tr><td> Status </td><td> Draft </td></tr>
+<!-- <tr><td> Status </td><td> Draft | Proposed | Accepted | Rejected | Deferred | Implemented </td></tr> -->
+<tr><td> Author(s) </td><td> Eric Dill &lt;mail@ericdill.dev&gt;</td></tr>
+<tr><td> Created </td><td> March 28, 2023</td></tr>
+<tr><td> Updated </td><td> NA </td></tr>
+<tr><td> Discussion </td><td> NA </td></tr>
+<tr><td> Implementation </td><td> NA </td></tr>
+</table>
+
+## Abstract
+
+This CEP aims to formalize the current state of the environment.yaml specification.
+It notably does not attempt to address any proposed changes to the specification.
+That is currently being discussed elsewhere.
+
+## Motivation
+
+There are a number of existing projects that can accept the env.yml format as input.
+Each project that wants to use the env.yml spec has to implement their own parser.
+This is a problem because it means that there are multiple implementations of the same specification.
+This makes it difficult to evolve the specification because right now it is implicitly defined as however `conda-env` handles the env.yml file.
+This CEP aims to formalize the specification so that it can be used as a reference for other projects, including `conda-env`.
+There are [existing docs](https://conda.io/projects/conda/en/latest/user-guide/tasks/manage-environments.html) that defines how a user should create an environment.yml file.
+From what I can tell this is the only formal specification for this file.
+It is the objective of this CEP to formalize the existing specification so that other projects can standardize on what they expect as input.
+Evolutions to this spec are expected and welcome, but are out of the scope for the initial proposal.
+
+Issues found in the wild that result from env.yaml not having a published spec:
+- [vscode issue #280](https://github.com/microsoft/vscode-python/issues/280)
+
+## Specification
+
+There are four keys that the conda env spec expects:
+
+`name`, `str`, specifying the name of the conda env that will be created.
+
+`channels`, `list`, specifying the channels that the solver should find and pull packages from
+
+`dependencies`, `list`, specifying the top-level dependencies that the solver should start with
```

Comment:
Should we be a bit more prescrptive in the structure of what the strings are that are accepted?

Reply:
I agree. Pehaps [this line](https://github.com/conda/conda/blob/a8e441e3c0e80b0d4e1595579f7d9eaad2b0fb2b/conda/models/match_spec.py#L92) from the matchspec source is sufficient

```

(channel(/subdir):(namespace):)name(version(build))[key1=value1,key2=value2]
```

Reply:
frankly i had no idea the env spec supported such a rich way to specify the specific package you wanted 

```
    Examples:
        >>> str(MatchSpec(name='foo', build='py2*', channel='conda-forge'))
        'conda-forge::foo[build=py2*]'
        >>> str(MatchSpec('foo 1.0 py27_0'))
        'foo==1.0=py27_0'
        >>> str(MatchSpec('foo=1.0=py27_0'))
        'foo==1.0=py27_0'
        >>> str(MatchSpec('conda-forge::foo[version=1.0.*]'))
        'conda-forge::foo=1.0'
        >>> str(MatchSpec('conda-forge/linux-64::foo>=1.0'))
        "conda-forge/linux-64::foo[version='>=1.0']"
        >>> str(MatchSpec('*/linux-64::foo>=1.0'))
        "foo[subdir=linux-64,version='>=1.0']"
```